providers/dns/constellix/constellix.go

```diff
@@ -0,0 +1,223 @@
+// Package constellix implements a DNS provider for solving the DNS-01 challenge using Constellix DNS.
+package constellix
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-acme/lego/v3/challenge/dns01"
+	"github.com/go-acme/lego/v3/platform/config/env"
+	"github.com/go-acme/lego/v3/providers/dns/constellix/internal"
+)
+
+// Config is used to configure the creation of the DNSProvider
+type Config struct {
+	APIKey             string
+	SecretKey          string
+	PropagationTimeout time.Duration
+	PollingInterval    time.Duration
+	TTL                int
+	HTTPClient         *http.Client
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider
+func NewDefaultConfig() *Config {
+	return &Config{
+		TTL:                env.GetOrDefaultInt("CONSTELLIX_TTL", 300),
+		PropagationTimeout: env.GetOrDefaultSecond("CONSTELLIX_PROPAGATION_TIMEOUT", dns01.DefaultPropagationTimeout),
+		PollingInterval:    env.GetOrDefaultSecond("CONSTELLIX_POLLING_INTERVAL", dns01.DefaultPollingInterval),
+		HTTPClient: &http.Client{
+			Timeout: env.GetOrDefaultSecond("CONSTELLIX_HTTP_TIMEOUT", 30*time.Second),
+		},
+	}
+}
+
+// DNSProvider is an implementation of the challenge.Provider interface.
+type DNSProvider struct {
+	config *Config
+	client *internal.Client
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for Constellix.
+// Credentials must be passed in the environment variables:
+// CONSTELLIX_API_KEY and CONSTELLIX_SECRET_KEY.
+func NewDNSProvider() (*DNSProvider, error) {
+	values, err := env.Get("CONSTELLIX_API_KEY", "CONSTELLIX_SECRET_KEY")
+	if err != nil {
+		return nil, fmt.Errorf("constellix: %w", err)
+	}
+
+	config := NewDefaultConfig()
+	config.APIKey = values["CONSTELLIX_API_KEY"]
+	config.SecretKey = values["CONSTELLIX_SECRET_KEY"]
+
+	return NewDNSProviderConfig(config)
+}
+
+// NewDNSProviderConfig return a DNSProvider instance configured for Constellix.
+func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
+	if config == nil {
+		return nil, errors.New("constellix: the configuration of the DNS provider is nil")
+	}
+
+	if config.SecretKey == "" || config.APIKey == "" {
+		return nil, errors.New("constellix: some credentials information are missing")
```

Comment:
Same remark as [here](https://github.com/go-acme/lego/pull/1056#discussion_r384083139): as a user, I'd find "something's missing" not particular helpful (i.e. what exactly is missing?).